apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1247,6 +1261,134 @@ spend(Arg0, Arg1, ES0) ->
+                  %% if the call was protected, rewrap the result
+                  case Protected of
+                      unprotected ->
+                          {{tuple, {}}, ES1_} = get_op_arg({stack, 0}, ES0_),
+                          write({stack, 0}, ?FATE_CONTRACT(ContractPK), ES1_);
+                      protected ->
+                          case get_op_arg({stack, 0}, ES0_) of
+                              {{variant,[0,1],1,{{tuple,{}}}}, ES1_} ->
+                                  write( {stack, 0}, make_some(?FATE_CONTRACT(ContractPK))
+                                       , ES1_
+                                       );
+                              {{variant, [0,1],0,{}}, ES1_} ->
+                                  write({stack, 0}, make_none(), ES1_)
+                          end
+                  end
+          end,
+    case remote_call_common(
+           ?FATE_CONTRACT(ContractPK),
+           init,
+           InitArgsTypes,
+           ?FATE_TYPEREP({tuple, []}),
+           0,
+           GasCap,
+           Protected,
+           ReplaceInitResult,
+           ES1)
+    of
+        {ok, ES2} ->
+            {jump, 0, ES2};
+        {failed_protected_call, ES2} ->
+            ES3 = aefa_engine_state:remove_contract(ContractPK, ES2),
+            {next, write({stack, 0}, make_none(), ES3)}
+    end.
+
+create_contract(CodeOrPK, Amount, ES0) ->
+    Current = aefa_engine_state:current_contract(ES0),
+
+    AS0              = aefa_engine_state:chain_api(ES0),
+    {ok, Nonce, AS1} = aefa_chain_api:next_nonce(Current, AS0),
+
+    Contract0  =
+        case CodeOrPK of
+            {new, Code} -> aect_contracts:new(
+                              Current, Nonce,
+                              #{vm => aefa_engine_state:vm_version(ES0)
+                              , abi => ?ABI_FATE_SOPHIA_1
+                              }, Code, 0);
+            {clone, PK} -> aect_contracts:new_clone(
+                             Current, Nonce,
+                             #{vm => aefa_engine_state:vm_version(ES0)
+                             , abi => ?ABI_FATE_SOPHIA_1
+                             }, PK, 0)
+        end,
+    Store      = aefa_stores:initial_contract_store(),
+    Contract1  = aect_contracts:set_state(Store, Contract0),
+    ContractPK = aect_contracts:pubkey(Contract1),
+
+    AS2 = aefa_chain_api:put_contract(Contract1, AS1),
+    {ok, AS3} = aefa_chain_api:transfer_value(Current, ContractPK, Amount, AS2),
+
+    ES1 = aefa_engine_state:set_chain_api(AS3, ES0),
+    {ContractPK, ES1}.
+
+bytecode_hash(Arg0, Arg1, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {?FATE_CONTRACT(Pubkey), ES1} = get_op_arg(Arg1, ES0),
+    case aefa_engine_state:contract_fate_bytecode(Pubkey, ES1) of
+        {ok, ByteCode, ?VM_FATE_SOPHIA_2, ES2} ->
+            SerByteCode = aeb_fate_code:serialize(ByteCode),
+            Hashed  = aeb_fate_data:make_hash(aec_hash:hash(fate_code, SerByteCode)),
```

Comment:
Minor concern. Should the (gas) price not depend somehow on the hashed contract bytecode size? What if somebody makes a big contract and keeps asking for its bytecode hash...

Reply:
Keeping a local cache at the node would also mitigate this

Reply:
Caching it would solve the problem I guess.

Reply:
I think it should have a dynamic cost for the size, it is done for other functions/operations!

Reply:
Now it is linear to the size of the hashed contract with coef=1. May be a bit too low, doesn't it?